### ISSUE TITLE
Reverse proxy URL base path

### ISSUE THREAD

**Issue by Froghut** [Ylianst/MeshCentral#239]
I want to set up nginx reverse proxy for mesh central but with a different URL base, let's say "/meshcentral" instead of "/".
If I change the "location" in nginx config from "/" to "/meshcentral" the site starts to load but will return 404 for all scripts etc that need to get loaded since meshcentral has "/script.js" instead of "/meshcentral/script.js" in the returned html.
Is there a setting  to adjust the url base path?

**Comment by Ylianst** [comment]
Hi. One easy way to do this is to create a second domain called "meshcentral" like this:

```
{
  "settings": {
    "Cert": "myserver.com",
    "MongoDb": "mongodb://127.0.0.1:27017/meshcentral"
  },
  "domains": {
    "": {
      "_title": "this is a test",
      "_title2": "unused domain"
    },
    "meshcentral": {
      "title": "TestServer",
      "title2": "Test",
      "newAccounts": 1,
      "footer": "test"
    }
  }
}
```

This way, when you hit /meshcentral on the server it will work. You can create more servers with new domains for /customer1 and /customer2...

Hope it helps,
Ylian

**Comment by Froghut** [comment]
I can now connect via the webui and agents can also connect to the server, but Desktop, Terminal and Files just hang in the connect state. Here is my nginx config in case there's something wrong there:

`location /meshcentral {
	proxy_send_timeout 999999s;
	proxy_read_timeout 999999s;
	proxy_pass http://127.0.0.1:4444/meshcentral;
	proxy_http_version 1.1;
	
	proxy_set_header Upgrade $http_upgrade;
	proxy_set_header Connection "upgrade";
	# Inform MeshCentral about the real host, port and protocol
	proxy_set_header X-Forwarded-Host $host:$server_port;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Proto $scheme;
}`

**Comment by Ylianst** [comment]
Interesting. Thanks for the report and NGNIX config. I think I have an idea about that could be going on. I will do some testing and make fixes as needed. Will do today or tomorrow.

**Comment by Ylianst** [comment]
Just published MeshCentral v0.3.4-z with a fix for this. The agent was connecting using https://server/meshrelay.ashx instead of https://server/domain/meshrelay.ashx causing this issue. Browser side was connecting using the correct URL. Please try and close the issue if it works - Thanks.

**Comment by Froghut** [comment]
Thank you, it works now! 
I found two small issues though, the website still requests "/commander.htm", ignoring the base path. And when using the setting "minify" the images to the left (leftbar-64.png etc) show the same behaviour.
And I guess this is more an issue of firefox, but it also tries to load "favicon.ico" ignoring the base path (I think because the website does not specify one itself), perhaps it would be nice to have an option to set a custom favicon to prevent this?

**Comment by Ylianst** [comment]
Excellent reports! I will fix all of these and will post back when the new version is published.

**Comment by Ylianst** [comment]
Just published MeshCentral v0.3.5-b with a fix for this.  Hopefully the favicon, command and missing images should not work. Please test and let me know.

**Comment by Froghut** [comment]
everything works now except the Backup, that still points to /Backup.zip

**Comment by Ylianst** [comment]
Ha! Again, thanks for reporting these! I will fix that.

**Comment by Ylianst** [comment]
I just fixed /Backup.zip, it will be in the next release.

**Comment by Froghut** [comment]
The link now points to the correct url, but when I try to download the server backup I get the following message: 
> Cannot GET /meshcentral/backup.zip

**Comment by Ylianst** [comment]
Thanks for the report. Published MeshCentral v0.3.5-f with a fix for /meshcentral/backup.zip.

**Comment by Froghut** [comment]
Perfect, everything works now, thanks for your fast help!
